p4-16/psa/PSA.mdk

```diff
@@ -309,6 +310,134 @@ Additional supported match_kind types
 ```
 
 
+## Data plane vs. control plane data representations
+
+A PSA data plane implementation that supports the P4 Runtime
+API[^P4RuntimeAPI] is expected to include software called a "P4
+Runtime API agent" that encapsulates details specific to the PSA
+implementation, enabling a P4 Runtime API controller to more easily
+control multiple devices with different PSA implementations.  We will
+refer to these pieces of software as an "agent" and "controller" for
+brevity.
+
+As mentioned in section [#sec-psa-type-definitions], different PSA
+implementations are expected to customize the size of the data types
+that refer directly to those objects, i.e. ports, multicast group ids,
+etc.
+
+Some PSA implementations are expected to use noticeably fewer
+resources for things like table keys and action parameters if the data
+plane stores only the fewest number of bits required for values of
+these types, for that implementation.  For example, an implementation
+that defines `PortId_t` as `bit<6>` instead of `bit<16>`, and can take
+advantage of this difference, could save 10 Mbits of storage in a
+table with a million entries[^CacheCost].
+
+[^CacheCost]: While 10 Mbits sounds tiny to one accustomed to computers with hundreds of gigabytes of DRAM, the highest speed PSA implementations are ASICs that must keep tables in on-chip memories, similar to caches in general purpose CPUs.  The Intel i9-7980XE released in 2017 has 198 Mbits of on-chip L3 cache shared by its CPU cores.  Among Intel processors in Intel's 7th generation Core released in 2017 with at least 100 Mbits of L3 cache, they all cost close to $9 per Mbit of L3 cache.  https://en.wikipedia.org/wiki/List_of_Intel_microprocessors
+
+The P4 Runtime API is planning to use 64-bit quantities to hold values
+of the types listed in section [#sec-psa-type-definitions].  For
+control plane operations on tables, any trimming or padding of values
+will be performed in the agent (trimming from controller to device
+direction, padding in device to controller direction).
+
+There are multiple channels of communication over which such values
+might be carried between the controller and the data plane.  These
+channels of communication include:
+
++ Control plane operations on tables, where values of these types may
+  be included as part of the key, or as an action parameter.
++ Packets sent to the CPU ("packet in" from the controller's
+  perspective), or received from the CPU ("packet out" from the
+  controller's perspective).
++ Fields in a `Digest` extern notification message (section
+  [#sec-packet-digest]).
++ Fields in the data contents of a `Register` array (section
+  [#sec-registers]).
++ Match fields of an entry in a `ValueSet` extern instance (section
```

Comment:
Is this still applicable given that we were expecting to remove value_set from PSA 1.0?

Reply:
good catch.  Removing that from this PR.